pkg/plugins/manager/client/client.go

```diff
@@ -40,6 +40,8 @@ func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest)
 		return nil, plugins.ErrPluginNotRegistered.Errorf("%w", backendplugin.ErrPluginNotRegistered)
 	}
 
+	ctx = context.WithValue(ctx, instrumentation.PluginInfo{}, plugin.JSONData.Info)
```

Comment:
I think it would be nicer if we have a specific instrumentation struct that has only the fields it needs to instrument. WDYT?